### PR TITLE
Update _eval_simplify for Xor

### LINKED ISSUE #25827: Exclusive or (XOR) of a bool symbol and its negation should return True
When using `simplify` we obtain the following basic simplications:
```
>>> from sympy import *
>>> a = symbols('a') 
>>> simplify(a&~a) 
False
>>> simplify(a|~a) 
True
>>> simplify(a^~a) 
a ^ ~a
```
But it is straightforward that `simplify(a^~a) == True` regardless of `a` being `True` or `False`.

Digging a bit into `sympy\logic\boolalg.py` it seems that the function `Xor` does not recognize the immediate simplification either, despite evaluating `a^a` correctly:
```
>>> from sympy import *
>>> a = symbols('a') 
>>> Xor(a, ~a)       
a ^ ~a
>>> Xor(a,a)           
False
```

### PR DESCRIPTION
**Pull request by Kwanky-Millet**
#### References to other Issues or PRs

Fixes #25827

#### Brief description of what is fixed or changed
The behaviour of `_eval_simplify` for Xor has been changed to simplify to `BooleanTrue` in case of (a ^ ~a).

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/logic/boolalg.py [hunk 1]
```diff
@@ -662,7 +680,7 @@ def _eval_simplify(self, **kwargs):
                     if (e.lhs != x or x in e.rhs.free_symbols) and x not in reps:
                         try:
                             m, b = linear_coeffs(
-                                Add(e.lhs, -e.rhs, evaluate=False), x)
+                                e.rewrite(Add, evaluate=False), x)
```
Comment (oscarbenjamin) [inline sympy/logic/boolalg.py:683]: This seems to be reverting changes that were recently made in master but that are unrelated to the PR.

Comment (Kwanky-Millet) [reply sympy/logic/boolalg.py]: Sorry. That was my carelessness. New to open sourcing.

File: sympy/logic/boolalg.py [hunk 2]
```diff
@@ -1048,6 +1048,9 @@ def _eval_simplify(self, **kwargs):
         # And and Or, we only simplify the partial expressions before using
         # patterns
         rv = self.func(*[a.simplify(**kwargs) for a in self.args])
+        if type(super()._eval_simplify(**kwargs)) == BooleanTrue:
```
Comment (asmeurer) [inline sympy/logic/boolalg.py:1051]: Why is this type check needed?

Comment (Kwanky-Millet) [reply sympy/logic/boolalg.py:1051]: If replaced directly, it'd write other simplifications in terms of And and Or.

Comment (asmeurer) [reply sympy/logic/boolalg.py:1051]: I see. A better way to write this would be

```py
rv = super()._eval_simplify(**kwargs)
if rv in [True, False]:
    return rv
```

Another option would be to add something to the pattern simplifications to handle this specific case in Xor, since otherwise we are just wasting a call to simplify_logic.

Comment (Kwanky-Millet) [reply sympy/logic/boolalg.py:1051]: You're right. I'll try it.
